rust/crd/src/lib.rs

```diff
@@ -128,80 +133,74 @@ impl TrinoRole {
+    pub fn rolegroup_ref(
+        &self,
+        trino: &TrinoCluster,
+        group_name: impl Into<String>,
+    ) -> RoleGroupRef<TrinoCluster> {
+        RoleGroupRef {
+            cluster: ObjectRef::from_obj(trino),
+            role: self.to_string(),
+            role_group: group_name.into(),
+        }
     }
 }
 
-impl HasCommands for TrinoCluster {
-    fn get_command_types() -> Vec<ApiResource> {
-        vec![
-            Start::api_resource(),
-            Stop::api_resource(),
-            Restart::api_resource(),
-        ]
+impl From<String> for TrinoRole {
+    fn from(item: String) -> Self {
+        if item == Self::Coordinator.to_string() {
+            Self::Coordinator
+        } else {
+            Self::Worker
+        }
     }
 }
 
-impl HasOwned for TrinoCluster {
-    fn owned_objects() -> Vec<&'static str> {
-        vec![Restart::crd_name(), Start::crd_name(), Stop::crd_name()]
-    }
-}
+#[derive(Clone, Debug, Default, Deserialize, JsonSchema, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TrinoClusterStatus {}
 
-impl HasApplication for TrinoCluster {
-    fn get_application_name() -> &'static str {
-        APP_NAME
-    }
+// TODO: move to operator-rs? Used for hive, opa, zookeeper ...
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ClusterRef {
+    pub name: String,
+    pub namespace: String,
+    pub chroot: Option<String>,
 }
 
-impl HasClusterExecutionStatus for TrinoCluster {
-    fn cluster_execution_status(&self) -> Option<ClusterExecutionStatus> {
-        self.status
-            .as_ref()
-            .and_then(|status| status.cluster_execution_status.clone())
-    }
-
-    fn cluster_execution_status_patch(&self, execution_status: &ClusterExecutionStatus) -> Value {
-        json!({ "clusterExecutionStatus": execution_status })
-    }
+// TODO: move to operator-rs? Copied from hive operator.
+/// Contains all the required connection information for S3.
+#[derive(Clone, Debug, Default, Deserialize, Eq, Hash, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct S3Connection {
+    pub end_point: String,
+    pub access_key: String,
+    pub secret_key: String,
+    pub ssl_enabled: bool,
+    pub path_style_access: bool,
 }
 
-#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[derive(Clone, Debug, Default, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct TrinoConfig {
     // config.properties
-    pub coordinator: Option<bool>,
-    pub http_server_http_port: Option<u16>,
-    pub http_server_https_port: Option<u16>,
     pub query_max_memory: Option<String>,
     pub query_max_memory_per_node: Option<String>,
     pub query_max_total_memory_per_node: Option<String>,
     // node.properties
     pub node_data_dir: Option<String>,
     // log.properties
     pub io_trino: Option<String>,
```

Comment:
Im not sure i get what you mean here?

Reply:
Reading only the object itself and the CRD definition (but not the Rust sources), what do you expect the following to do?

```yaml
config:
  ioTrino: foo
```

Reply:
It will throw an error in the product config:
```
  - property: &ioTrino
      propertyNames:
        - name: "io.trino"
          kind:
            type: "file"
            file: "log.properties"
      datatype:
        type: "string"
      defaultValues:
        - fromVersion: "0.0.0"
          value: "INFO"
      allowedValues:
        - "INFO"
        - "DEBUG"
        - "WARN"
        - "ERROR"          
      roles:
        - name: "coordinator"
          required: true
        - name: "worker"
          required: true
      asOfVersion: "0.0.0"
```

But yeah probably makes sense to get this into an enum.

Reply:
I was more concerned about renaming it to something like `logLevel`.

Reply:
that makes sense too, done ;)